inc/template.php

```diff
@@ -170,6 +191,8 @@ function set_authors( WP_Post $post, array $authors ) : array {
 		throw new Exception( $terms->get_error_message() );
 	}
 
+	wp_cache_flush();
```

Comment:
I think flushing the _entire_ cache is too heavy handed. Remember this is a plugin used by many sites! As much as cache _should_ be ephemeral, I think we have to be cautious about clearing it. Take for example a large high traffic site with a _lot_ of cached data, clearing it when a new post is published could significantly impact the performance for subsequent requests. 

It's easy to get around though. Just delete the individual cache entry by key. 

```suggestion
	wp_cache_delete( 'author_ids_' . $post->ID, 'authorship' );
```